src/transformers/generation/utils.py

```diff
@@ -351,6 +359,43 @@ def prepare_inputs_for_generation(self, *args, **kwargs):
             "A model class needs to define a `prepare_inputs_for_generation` method in order to use `.generate()`."
         )
 
+    def set_cache_implementation(self, cache_implementation: Union[CacheImplementation, str], **kwargs):
+        """
+        Simple API to switch cache implementation in a model. Users could also do
+        `model.config.generation_config.cache_implementation = xxx` but they will most likely unexpected
+        behaviour
+
+        Args:
+            cache_implementation (`Union[CacheImplementation, str]`):
+                The target cache implementation
+            kwargs (`dict`, *optional*):
+                Optional key word arguments to be passed. E.g. for SinkCache, it is required to
+                pass `window_length` and `num_sink_tokens`.
+        """
+        if cache_implementation.upper() not in CacheImplementation.__members__:
+            raise ValueError(
+                f"Unrecognized cache implementation - you passed {cache_implementation}. Supported cache implementations are"
+                f" {CacheImplementation.__members__}"
+            )
+
+        if not self._supports_cache_class:
+            raise ValueError(
+                "This model do not currently support switching between cache implementations. Please raise a feature request on GitHub for adding"
+                " the request to support this model: https://github.com/huggingface/transformers"
```

Comment:
```suggestion
                "You cannot currently update the cache implementation for this model. Please raise a feature request on GitHub for adding"
                " improved cache support this model: https://github.com/huggingface/transformers"
```
Maybe something like that?